lib/dfs.js

```diff
@@ -29,26 +29,37 @@ class DFS {
     return { traverseOrder, missing };
   }
 
-  traverseAndBundle(node, getAdjacentAndBundle) {
+  async traverseAndBundle(node, getAdjacentAndBundle) {
     let traverseOrder = [],
       stack = [],
       missing = [],
       visited = new Set(),
       nodeContents = {},
-      globalReferences = {};
+      globalReferences = {},
+      hrefsVisited = new Set();
+
     stack.push(node);
     while (stack.length > 0) {
       node = stack.pop();
-      if (!visited.has(node)) {
+      if (!visited.has(node) &&
+
+        /**
+         * For nodes that are fetched for remote URLs we ensure they
+         * aren't visited more than once
+         */
+        (!node.href || (!hrefsVisited.has(node.href)))
```

Comment:
In which cases `!node.href` can be true? i.e. if `node.href` is empty, in subsequent logic line no. 54 here, we're adding it to `hrefsVisited`. Not sure if this is intended.

Reply:
!node.href would be true for all non remote reference nodes, I'll add an extra check at line 54